src/EditorFeatures/CSharp/SignatureHelp/InvocationExpressionSignatureHelpProvider_MethodGroup.cs

```diff
@@ -43,7 +43,8 @@ internal partial class InvocationExpressionSignatureHelpProvider
                 }
 
                 var includeInstance = !throughExpression.IsKind(SyntaxKind.IdentifierName) ||
-                    semanticModel.LookupSymbols(throughExpression.SpanStart, name: throughSymbol.Name).Any(s => !(s is INamedTypeSymbol));
+                    semanticModel.LookupSymbols(throughExpression.SpanStart, name: throughSymbol.Name).Any(s => !(s is INamedTypeSymbol)) ||
+                    semanticModel.LookupSymbols(throughExpression.SpanStart, container: throughSymbol.ContainingType).Any(s => !(s is INamedTypeSymbol));
```

Comment:
The crash occurred on the next line, `Contract.ThrowIfFalse()` that was unhandled, and this makes the boolean true where it should be. As for ContainingType, the default to the method is null anyways and that simply means 'from this context', similar to the line above that doesn't specify the container, so it defaults to the class containing the method, `B` in this case.